internal/n/reliable_notifier.go

```diff
@@ -92,7 +92,9 @@ func runEntrypointListener(
 				)
 				select {
 				case <-notifyCtx.Done():
-					settings.onNotifierTimeout(name)
+					// execute this callback on a goroutine to not hang this notification
+					// system
+					go settings.onNotifierTimeout(name)
```

Comment:
No risk of leaking?

Reply:
Sigh, there is only bad outcomes here, so we need to pick, what should we prioritize? hanging the whole notification system because the callback may hang, or leaking goroutines temporarily until the callback finishes

Reply:
The main thing I would want to avoid (in addition to not leaking go routines if we can avoid it) is to prevent one notifier delaying another. If there were a way to separate them into different streams that can be processed independently (maybe another layer of channels? 😬 or just remove `onNotifierTimeout` and log it instead).

Reply:
You know, most problems can be addressed with another layer of abstraction. The unfortunate corollary is that most problems can be overcomplicated by another layer of abstraction. I am tempted to err with the current set up and learn.

Reply:
Personally, I prefer to not do the goroutine and add a warning in the documentation saying that your callback must not block. That scenario is not going to happen to us given we are going to use Prometheus non-blocking API.

Reply:
I'm ok with that tradeoff 👍